avm/res/compute/disk-encryption-set/modules/nested_keyVaultPermissions.bicep

```diff
@@ -0,0 +1,69 @@
+@description('Required. A boolean to specify whether or not the used Key Vault has RBAC authentication enabled or not.')
+param rbacAuthorizationEnabled bool = true
+
+@description('Required. The resourceID of the User Assigned Identity to assign permissions to.')
+param userAssignedIdentityResourceId string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Required. Resource ID of the KeyVault containing the key or secret.')
+param keyVaultResourceId string
+
+@description('Required. Key URL (with version) pointing to a key or secret in KeyVault.')
+param keyName string
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-10-01' existing = {
+  name: last(split(keyVaultResourceId, '/'))!
+
+  resource key 'keys@2021-10-01' existing = {
+    name: keyName
+  }
+}
+
+module userAssignedIdentity 'nested_managedIdentityReference.bicep' = {
```

Comment:
Can this not just be an existing resource? Straight in this file?

Reply:
Yes, it can. Implemented:
- https://github.com/Azure/bicep-registry-modules/pull/749/commits/2e20017846bb85638d18183b71051fbe7f286dc2
- https://github.com/Azure/bicep-registry-modules/pull/749/commits/d61f2a4698e61d655f3055c7e49966920db4f0c5